src/components/round/SearchInput.vue

```diff
@@ -47,33 +47,35 @@ function handleSelectionMovement(event: KeyboardEvent) {
   if (event.key === 'ArrowDown' || event.key === 'ArrowUp') {
     event.preventDefault()
 
-    if (selectedTrack.value === undefined) {
+    const moveDown = event.key === 'ArrowDown'
+    const moveUp = event.key === 'ArrowUp'
+    const selectedTrackExistsAndIsNotLast = selectedTrack.value && matchingTracks.value.indexOf(selectedTrack.value) < matchingTracks.value.length - 1
+    const selectedTrackExistsAndIsNotFirst = selectedTrack.value && matchingTracks.value.indexOf(selectedTrack.value) > 0
+
+    if (!selectedTrack.value) {
       selectedTrack.value = matchingTracks.value[0]
     }
-    else if (
-      event.key === 'ArrowDown'
-      && selectedTrack.value !== undefined
-      && matchingTracks.value.indexOf(selectedTrack.value)
-      < matchingTracks.value.length - 1
-    ) {
+    else if (moveDown && selectedTrackExistsAndIsNotLast) {
       const newIndex = matchingTracks.value.indexOf(selectedTrack.value) + 1
       selectedTrack.value = matchingTracks.value[newIndex]
     }
-    else if (
-      event.key === 'ArrowUp'
-      && selectedTrack.value !== undefined
-      && matchingTracks.value.indexOf(selectedTrack.value) > 0
-    ) {
+    else if (moveUp && selectedTrackExistsAndIsNotFirst) {
       const newIndex = matchingTracks.value.indexOf(selectedTrack.value) - 1
       selectedTrack.value = matchingTracks.value[newIndex]
     }
   }
 }
 
 function handleSelectionInput(event: KeyboardEvent) {
-  if (event.key === 'Enter' && selectedTrack.value !== undefined && focused.value && matchingTracks.value.length !== 0) {
+  const inputIsFocused = focused.value
+  const clickedEnter = event.key === 'Enter'
+  const selectedAnyTrack = selectedTrack.value
+  const selectedTrackMatchesGuess = matchingTracks.value.length !== 0
+  const selectedVisibleTrack = selectedAnyTrack && selectedTrackMatchesGuess
```

Comment:
```ts
const selectedVisibleTrack = selectedTrack.value && matchingTracks.value.length !== 0
```

Reply:
re: możę selectedCorrectTrack zamiast visible? chyba ze to w kontekscie apki ma wiekszy sens i ze faktycznie ktos kliknal w widoczny track